src/server/components/lists/controllers/contactUsController.ts

```diff
@@ -0,0 +1,114 @@
+  email: "Enter your email address",
+  name: "Enter your name",
+  providerCompanyName: "What is their company name? (Optional)",
+  providerName: "What is the service provider's name",
+  serviceType: "What service type are you contacting us about?",
+};
+
+export function getComplaintForm(req: Request, res: Response) {
+  const [errors] = req.flash("errors") as unknown as string[];
+  const errorList = errors ? (JSON.parse(errors) as Array<Record<string, string>>) : null;
+
+  if (!errorList) {
+    res.render("help/contact-us", { csrfToken: getCSRFToken(req), fieldTitles, countriesList });
+    return;
+  }
+
+  const errorsObj = errorList.reduce(
+    (acc, error) => ({
+      ...acc,
+      [error.key]: error.text,
+    }),
+    {}
+  ) as Partial<ContactUsFormFields>;
+
+  res.render("help/contact-us", {
+    csrfToken: getCSRFToken(req),
+    fieldTitles,
+    countriesList,
+    errors: errorsObj,
+    errorList,
+  });
+}
+
+export async function postComplaintForm(req: Request, res: Response, next: NextFunction) {
+  const ERROR_MESSAGES = {
+    "string.empty": "{{#label}} is required",
+  };
+
+  const contactUsFormSchema = Joi.object({
+    country: Joi.string().label(fieldTitles.country).required().messages(ERROR_MESSAGES),
+    detail: Joi.string().label(fieldTitles.detail).required().messages(ERROR_MESSAGES),
+    email: Joi.string().email().label(fieldTitles.email).required().messages(ERROR_MESSAGES),
+    name: Joi.string().label(fieldTitles.name).required().messages(ERROR_MESSAGES),
+    providerCompanyName: Joi.string().allow(null, ""),
+    providerName: Joi.string().label(fieldTitles.providerName).required().messages(ERROR_MESSAGES),
+    serviceType: Joi.string().label(fieldTitles.serviceType).required().messages(ERROR_MESSAGES),
+  });
+
+  const { _csrf, ...formFields } = req.body as ContactUsFormFields;
+  const { value: validatedFormFields, error: validationError } = contactUsFormSchema.validate(formFields, {
+    abortEarly: false,
+    stripUnknown: true,
+    errors: {
+      wrap: {
+        label: false,
+      },
+    },
+  });
+
+  if (validationError) {
+    const errors = validationError.details.map((detail) => {
+      const { key } = detail.context!;
+      return {
+        text: detail.message,
+        href: `#${key}`,
+        key,
+      };
+    });
+    req.flash("errors", JSON.stringify(errors));
+    logger.error(`postComplaintForm Error: Validation failed - ${validationError.message}`);
+    res.redirect("/help/contact-us");
+    return;
+  }
+
+  try {
+    const personalisation = {
+      emailSubject: `A ${validatedFormFields.serviceType} in ${validatedFormFields.country}: Find service complaint form`,
+      emailPayload: formatFieldData(validatedFormFields).join("\r\n\n ## \r\n"),
+    };
+
+    await sendContactUsEmail(personalisation);
+    res.redirect("/help/contact-us-confirm");
+  } catch (error) {
+    logger.error(`postComplaintForm Error: ${error.errors ?? error.message}`);
```

Comment:
sorry, I didn't go through my code changes like I usually do. just an FYI if you didn't know, promise.any returns an AggregateError if lots of things failed. 

`AggregateError` has the prop `errors`. It doesn't return an error if at least 1 promise resolves.

Reply:
Ah okay that's useful to know.

I'm guessing the code you've highlighted is just for explanation and not because it needs to be changed?